contracts/smart-contract-wallet/test/SocialRecoveryModule.sol

```diff
@@ -0,0 +1,119 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.12;
+import "../SmartAccount.sol";
+
+contract SocialRecoveryModule {
+    string public constant NAME = "Social Recovery Module";
+    string public constant VERSION = "0.1.0";
+
+    struct Friends {
+        address[] friends; // the list of friends
+        uint256 threshold; // minimum number of friends required to recover
+    }
+    mapping(address => Friends) internal friendsEntries;
+    mapping(address => mapping(address => bool)) public isFriend;
+
+    // isConfirmed - map of [recoveryHash][friend] to bool
+    mapping(bytes32 => mapping(address => bool)) public isConfirmed;
+    mapping(address => uint256) internal walletsNonces;
+
+    /**
+     * @dev Setup function sets initial storage of contract. Only by SCW owner.
+     */
+    function setup(address[] memory _friends, uint256 _threshold) public {
```

Comment:
later we can update this with wallet address as argument and onlyModule(_wallet) modifier such that setUp is called after module is enabled. 
currently module can be enabled later which is ok as per this logic

Reply:
setup is called by SCW only as the friends is stored with the `msg.sender`, but got your point